scripts/jmwalletd.py

```diff
@@ -0,0 +1,626 @@
+    @app.route('/wallet/all', methods=['GET'])
+    def listwallets(self, request):
+        #this is according to the assumption that wallets are there in /.joinmarket by default, also currently path for linux system only.
+        #first user taken for path
+        user_path = glob.glob('/home/*/')[0]
+        
+        wallet_dir = f"{user_path}.joinmarket/wallets/*.jmdat"
+        wallets = (glob.glob(wallet_dir))
+        
+        offset = len(user_path)+len('.joinmarket/wallets/')
+        #to get only names
+        short_wallets = [wallet[offset:] for wallet in wallets]
+        return response(request,wallets=short_wallets)
+
+    #route to get external address for deposit
+    @app.route('/address/new/<string:mixdepth>',methods=['GET'])
+    def getaddress(self, request, mixdepth):
+        # self.check_cookie(request)
+        if not self.wallet_service:
+            raise NoWalletFound()
+        mixdepth = int(mixdepth)
+        address = self.wallet_service.get_external_addr(mixdepth)
+        return response(request,address=address)
+
+    #route to list utxos
+    @app.route('/wallet/transactions',methods=['GET'])
+    def listUtxos(self, request):
+        if not self.wallet_service:
+            raise NoWalletFound()
+        utxos = wallet_showutxos(self.wallet_service, False)
+        
+        return response(request,transactions=utxos)
+
+    #return True for now
+    def filter_orders_callback(orderfees, cjamount,dummy):
+        return True
+
+
+    #route to start a coinjoin transaction
+    @app.route('/wallet/taker/coinjoin',methods=['POST'])
+    def doCoinjoin(self, request):
+
+        if not self.wallet_service:
+            raise NoWalletFound()
+
+        request_data = self.get_POST_body(request,["mixdepth", "amount", "counterparties","destination"])
+        #refer sample schedule testnet
+        waittime = 0
+        rounding=16
+        completion_flag=0
+        #list of tuple
+        schedule = [(request_data["mixdepth"],request_data["amount"],request_data["counterparties"],request_data["destination"],waittime,rounding,completion_flag)]
+        print(schedule)
+        #instantiate a taker
+        #keeping order_chooser as default for now
+        taker = Taker(self.wallet_service, schedule, max_cj_fee = (1,float('inf')), callbacks=(self.filter_orders_callback, None,  self.taker_finished))
+
+        clientfactory = JMClientProtocolFactory(taker)
+        
+        nodaemon = jm_single().config.getint("DAEMON", "no_daemon")
+        daemon = True if nodaemon == 1 else False
+        dhost = jm_single().config.get("DAEMON", "daemon_host")
+        dport = jm_single().config.getint("DAEMON", "daemon_port")
+        
+        if jm_single().config.get("BLOCKCHAIN", "network") == "regtest":
+            startLogging(sys.stdout)
+        start_reactor(dhost, dport, clientfactory, daemon=daemon)
+
+    def taker_finished(res, fromtx=False, waittime=0.0, txdetails=None):
```

Comment:
I think there's an error here: the first argument should be `self`, it refers to the instance of the class.

Of course, changing its name to `res` not `self`, in itself, doesn't harm anything, except: this is a callback that partially responds to the first argument's value (it's used to indicate success or failure of the transaction attempt, if I recall correctly).

I'm not 100% sure of the effect of what we have here, but there might be a case of the first argument getting 'swallowed'. Needs to be checked.